src/reynir_correct/pattern.py

```diff
@@ -178,15 +179,38 @@ def __init__(self, ann: List[Annotation], sent: Sentence) -> None:
     def get_wordform(self, lemma: str, cat: str, variants: Iterable[str]):
         """ Get correct wordform from BinPackage, 
         given a set of variants """
-
-        # Get rid of argument variants in verbs:
-        realvars = [v for v in variants if not v.isdigit() and v not in SKIPVARS]
+        realvars: Iterable[str]
+        if cat == "so":
+            # Get rid of argument variants in verbs:
+            realvars = frozenset([v for v in variants if not v.isdigit() and v not in SKIPVARS])
+            if not "lh" in realvars:
+                realvars = frozenset([v for v in realvars if v not in ALL_CASES])
+        else:
+            realvars= frozenset(variants)
         wordforms = BIN.lookup_variants(lemma, cat, tuple(realvars))
         if not wordforms:
             return ""
         # Can be many possible word forms, want the first one in most cases
         return wordforms[0].bmynd
 
+    def wrong_subtree(self, match: SimpleTree, so: SimpleTree) -> bool:
```

Comment:
We might consider changing the >> operator in the matcher logic so that it doesn't enter into IP subtrees. If this functionality is really needed, we might put it in a new >>> operator.

Reply:
Agree. Put in a TODO comment.